diff --git a/src/CKKS/ApproxModEval.cu b/src/CKKS/ApproxModEval.cu
index c0e300d..3caf9a1 100644
--- a/src/CKKS/ApproxModEval.cu
+++ b/src/CKKS/ApproxModEval.cu
@@ -380,7 +380,7 @@ const std::vector<double>& coefficients, double a, double b) const {
-			ctxt.addScalar(-lower_bound + 1.0);
+			ctxt.addScalar(-(upper_bound + lower_bound) / 2.0);
 		} else {
 			if (abs(lower_bound + upper_bound) > 1e-8)
-				ctxt.addScalar(-(upper_bound - lower_bound) / 2.0); // center on 0 (substract mean)
+				ctxt.addScalar(-(upper_bound + lower_bound) / 2.0); // center on 0 (subtract mean)
 			if (ctxt.cc.rescaleTechnique == CKKS::FIXEDMANUAL && ctxt.NoiseLevel == 2)
 				ctxt.rescale();
 			ctxt.multScalar(2.0 / (upper_bound - lower_bound));
